src/magellan_sim/src/MagellanSim.cpp

```diff
@@ -0,0 +1,68 @@
+#include "MagellanSim.h"
+
+MagellanSim::MagellanSim(ros::NodeHandle& nh) :
+        node_handle_(nh),
+        commanded_velocity_(0.0),
+        commanded_radius_(0.0),
+        velocity_(0.0),
+        steering_angle_(0.0),
+        refresh_rate_(10.0) {
+    throttle_subscriber_ =
+        node_handle_.subscribe("/platform/cmd_velocity", 1, &MagellanSim::UpdateThrottle, this);
+    steering_subscriber_ =
+        node_handle_.subscribe("platform/cmd_turning_radius", 1, &MagellanSim::UpdateSteering, this);
+    velocity_publisher_ =
+        node_handle_.advertise<geometry_msgs::TwistWithCovarianceStamped>("/platform/velocity", 1, true);
+    turning_radius_publisher_ =
+        node_handle_.advertise<std_msgs::Float64>("/platform/turning_radius", 1, true);
+    time_ = ros::Time::now();
+}
+
+void MagellanSim::run() {
+    while (node_handle_.ok()) {
+        ros::spinOnce();
+        Update();
+        ROS_INFO("Velocity: %0.2f\t Turning Radius: %0.2f", velocity_, commanded_radius_);
```

Comment:
Can you make this a ROS_DEBUG or a ROS_INFO_THROTTLE so that your terminal isn't spammed?

Reply:
Sure, will make it as ROS_DEBUG message